### PR TITLE
Added retry loop to _HandleMetadataUpdate

### PR DESCRIPTION
**Pull request by BongoEADGC6**
This is in response to issue #862 opened by @rjschwei

Added a retry counter with a default of 50 tries before breaking the loop.

### CODE REVIEW

**Comment by BongoEADGC6** [comment]
@hopkiw For your review

File: packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py [hunk 1]
```diff
@@ -178,6 +178,7 @@ def _HandleMetadataUpdate(
           exception = e
           self.logger.error('GET request error retrieving metadata. %s.', e)
         if retry:
+          retry_count += 1
           continue
         else:
           break
```
Comment (hopkiw) [inline packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py]:
```
if retries:
  retries -= 1
  time.sleep(1)
else:
  break
```

File: packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py [hunk 2]
```diff
@@ -154,21 +154,21 @@ def _GetMetadataUpdate(
 
   def _HandleMetadataUpdate(
       self, metadata_key='', recursive=True, wait=True, timeout=None,
-      retry=True):
+      retry=True, retry_limit=50):
```
[the remark this answers is not in the document]
Comment (BongoEADGC6) [reply packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py]: Understood. Let me dig in a bit.

File: packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py [hunk 3]
```diff
@@ -177,11 +176,8 @@ def _HandleMetadataUpdate(
         if not isinstance(e, type(exception)):
           exception = e
           self.logger.error('GET request error retrieving metadata. %s.', e)
-        if retry:
-          retry_count += 1
-          continue
-        else:
-          break
+        time.sleep(1)
+        continue
```
Comment (hopkiw) [inline packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py]: continue is not needed at end of loop. where are we decrementing the `retries` variable? won't this run in an infinite loop, now?

Comment (BongoEADGC6) [reply packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py]: Just fixed that all. Accidentally deleted.

**Comment by BongoEADGC6** [comment]
Okay... Think I'm done now.

**Comment by BongoEADGC6** [comment]
@hopkiw Hey there. Just waiting on review.

File: packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py [hunk 4]
```diff
@@ -154,33 +154,31 @@ def _GetMetadataUpdate(
...
       timeout: int, timeout in seconds for returning metadata output.
-      retry: bool, True if we should retry on failure.
+      retries: int, Number of times to retry obtaining metadata.
 
     Returns:
       json, the deserialized contents of the metadata server.
     """
     exception = None
-    while True:
+    while retries > -1:
```
Comment (hopkiw) [inline packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py]: `>=0` i think is easier to understand here

Comment (illfelder) [reply packages/python-google-compute-engine/google_compute_engine/metadata_watcher.py]: Friendly ping - can you please make this update?

**Comment by BongoEADGC6** [comment]
@hopkiw Thanks for finishing this out. I was away for a few weeks.

**Pull request merged**